cmd/root.go

```diff
@@ -272,3 +270,25 @@ func setupImagePullSecretsProvider() secrets.ImagePullSecretsProvider {
 
 	return secrets.NewKubernetesImagePullSecretsProvider(clientset)
 }
+
+// setupRegistry configures a target registry client connection
+func setupTargetRegistryClient() (registry.Client, error) {
+	targetRegistry, err := types.ParseTargetRegistry(cfg.Target.Type)
+	if err != nil {
+		log.Err(err)
+	}
+
+	switch targetRegistry {
+	case types.TargetRegistryGcp:
+		return registry.NewGARClient(cfg.Target.GCP.Location, cfg.Target.GCP.ProjectID, cfg.Target.GCP.RepositoryID)
```

Comment:
There are 2 return values expected. Is `nil` being implicitly passed?

Reply:
`registry.NewGARClient(...` returns `(*GARClient, error)`. I don't do any error checking in GCP case. There is error checking in the AWS case because we use the registry to perform `SetRepositoryTags` before returning.

I could update the GCP case to do error checking before returning if that is preferred. 

OR I could put all the `SetRepositoryTags` logic within `NewECRClient`, which I like the most at this time.

If there is a better or more preferred way, please let me know.

Reply:
The `SetRepositoryTags` functionality is now included when instantiating a `GARClient`. This simplifies the case logic.

Reply:
TIL that you can return `(*GARClient, error)` directly, cheers :)

As for `SetRepositoryTags`, I think I dislike both options. Moving  `SetRepositoryTags` into `NewECRClient` violates the separation of concern, but I do see the appeal.

With the proliferation of parameters on `NewECRClient` I was wondering if it makes sense to pass in the entire config object instead, as well as use a generic function like `Bootstrap` or `Init` to run things that need running after initialisation, e.g. `SetRepositoryTags` for ECR.

Let me know what you think.

Reply:
I have updated both new client functions (`NewECRClient` and `NewGARClient`) to use the config object (`config.AWS` and `config.GCP` respectively) as a single parameter.

The `SetRepositoryTags` was only taking the current tag configuration found in `config.AWS.ECROptions.Tags` and setting it on the `ECRClient`. I think setting `tags` on `ECRClient` within `NewECRClient` is consistent with how `accessPolicy` and `lifecyclePolicy` are also set within `NewECRClient`. I'm happy to change this, would you prefer all of these move to a generic function like `Bootstrap` or `Init`?

With the new update to pass the config object in I'm setting everything from the config object.